app/src/main/java/com/wireguard/android/fragment/TunnelListFragment.kt

```diff
@@ -206,10 +221,28 @@ class TunnelListFragment : BaseFragment() {
             tunnelList.addOnScrollListener(FloatingActionButtonRecyclerViewScrollListener(createFab))
             executePendingBindings()
         }
+        // Collapse searchview on fragment transaction
+        parentFragmentManager.addOnBackStackChangedListener {
+            if (searchItem.isActionViewExpanded) {
+                searchItem.collapseActionView()
+            }
+        }
         return binding?.root
     }
 
+    // Collapse searchview on activity change
+    override fun onPause() {
+        binding?.tunnels?.clear()
+        binding?.tunnels?.addAll(savedTunnelsList)
+        if (searchItem.isActionViewExpanded) {
+            searchItem.collapseActionView()
+        }
+        super.onPause()
+    }
+
     override fun onDestroyView() {
+        // Add back saved tunnels before setting it null since tunnel manager and binding are in sync.
+        binding?.tunnels?.addAll(savedTunnelsList)
```

Comment:
What's the downside of not doing this? Isn't the view being destroyed anyway? The binding is nullified in the very next line

Reply:
The fragment can go from onDestroyView() to onCreateView(), now the tunnelManager can have old tunnels if the user searched something and went to another fragment. This is just a saftey mechanism to make sure the tunnelManager is updated when fragment returns to layout from the back stack. Though another _maybe better_ solution can be updating binding to savedTunnelsList in onCreateView()

Reply:
I'd prefer setting this up in onCreateView instead, doing setup during destruction seems counterproductive.

Reply:
or we can modify the viewStateRestored event to not take tunnels from tunnelManager and use saved Tunnels

Reply:
Fine by me